extensions/dbgobject/core/dbgobject.js

```diff
@@ -855,9 +859,9 @@ Loader.OnLoad(function() {
                 };
                 return getRestOfString(that, "", /*suggestedChunkSize*/64);
             } else {
-                return that.vals(length)
+                return signedOrUnsignedVals(that, length)
                 .then(function (chars) {
-                    return chars.map(function (c) { return String.fromCharCode(c); }).join("");
+                    return chars.reduce((str, c) => str + String.fromCharCode(c), "");
```

Comment:
Isn’t this n^2?

Reply:
Isn't .join('') also n^2? I think it depends on how these things are implemented under the hood and what optimizations can be applied. (I can imagine .join being written to not be n^2, but perhaps so can .reduce.) I ran this perf comparison of map().join() vs. reduce(): http://jsbench.github.io/#a80c1e0e7282af6742adbbea4dfdf50a. I found that it was only when I increased strings sizes to ~500000 characters that map().join() was faster, otherwise reduce() performed better. So maybe .join's optimization are only helpful when we have VERY large strings.

To be clear though, I wasn't making this change for perf reasons. :)
Reduce seems cleaner to me, though I don't have a super strong opinion about it...

Reply:
Will pull this out of this change either way. If you think it is worth keeping, then I'll commit it separately.

Reply:
Yeah, this does not handle the Unicode case correctly for WTF::StringImpl (e.g. load http://www.google.com.hk with NG enabled and look at LayoutNGText::text_). But that was already broken, so could be fixed separately.

Reply:
Ah you're right...this doesn't work for UTF-16 sequences at all. Working on a fix - thanks!

Reply:
Both UTF-16 and Latin-1 should be handled correctly now.

Also String.fromCharCode can take a list of char codes, so we don't map().join() or reduce(). We can just spread the "chars" array.

Reply:
Why doesn't this (either what was here or what you have) work with UTF-16?

w.r.t. to reduce(...) vs. map(..).join(), I completely agree that reduce is cleaner -- my concern is just if there's a point at which the string concatentation explodes (1M+ characters isn't a ridiculous scenario.)  If you've convinced yourself that there's no such point, no objection.

Reply:
@petersalas It does work after fixing isUnsigned. Are you talking about something else?